packages/core/android/src/main/java/com/segmentanalyticsreactnative/AnalyticsReactNativeModule.kt

```diff
@@ -139,16 +148,17 @@ class AnalyticsReactNativeModule : ReactContextBaseJavaModule, ActivityEventList
 
     val contextInfo: WritableMap = Arguments.createMap()
 
-    // generate random identifier that does not persist across installations
-    // use it as the fallback in case DRM API failed to generate one.
-    val fallbackDeviceId = UUID.randomUUID().toString()
-    val deviceId = getUniqueId(config.hasKey("collectDeviceId") && config.getBoolean("collectDeviceId"))?: fallbackDeviceId
+    val deviceId = getUniqueId(config.hasKey("collectDeviceId") && config.getBoolean("collectDeviceId"))?: getFallbackDeviceId(config.hasKey("collectDeviceId") && config.getBoolean("collectDeviceId"))
```

Comment:
This seems too convoluted. Maybe just do a check if `config.hasKey("collectDeviceId")` and trigger the logic of getting the UniqueID or the fallback value. No need to repeat checking the config for every function.

Reply:
@oscb fair point. I just added another commit that works when i toggle the `collectDeviceId` config boolean. Let me know what you think